### DISCUSSION TITLE
Unable to build Ubuntu 24.04 - build guide needed

### DISCUSSION THREAD

**Discussion by avfxtd** [AcademySoftwareFoundation/OpenRV#482] [Build Issues]
Have seen that a few users managed to get this built on Ubuntu, however I am struggling.  Are there any plans for a build and install guide on Ubuntu based systems?  Cmake seems to be the stopper:

CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Qt5/Qt5Config.cmake:28 (find_package):
  Could not find a package configuration file provided by "Qt5WebEngineCore"
  with any of the following names:

    Qt5WebEngineCoreConfig.cmake
    qt5webenginecore-config.cmake

  Add the installation prefix of "Qt5WebEngineCore" to CMAKE_PREFIX_PATH or
  set "Qt5WebEngineCore_DIR" to a directory containing one of the above
  files.  If "Qt5WebEngineCore" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  cmake/dependencies/qt5.cmake:50 (FIND_PACKAGE)
  cmake/dependencies/CMakeLists.txt:48 (INCLUDE)

**Comment by cedrik-fuoco-adsk** [comment]
Hello @avfxtd!

Did you get your Qt installation from the Qt unified setup or from other source? Some other source might not provide QtWebEngine.

**Reply by avfxtd** [reply]
Thanks for responding.  Ok this could be the issue, I've not installed QT specifically.  New system I'm setting up.  Will check this and report back.

**Reply by avfxtd** [reply]
I've installed QT from the offical site now, and made sure that Qt WebEngine is also installed.  Still getting the same error.  I am installing this into a contained conda env as I was hesitant to make it system wide.  Not sure if thats potentially causing an issue.

![image](https://github.com/AcademySoftwareFoundation/OpenRV/assets/169622047/680759eb-ca48-418f-8126-53a006b57c54)

**Reply by geffrak** [reply]
You need Qt 5.15.y. Available from the archive section.

**Reply by avfxtd** [reply]
Ok thanks.  Do I only need the Qt Webengine for 5.15?  Its quite a lot to install otherwise

**Reply by geffrak** [reply]
Download the last version of Qt 5.15.x that you can get using the online installer on the Qt page. Logs, Android, iOS and WebAssembly are not required to build OpenRV.

You can remove your installation of Qt 6 for now.

**Reply by avfxtd** [reply]
Thanks.  That got much further with the install.  Slowly working through missing dependencies.  

This one I'm not sure about, it fails after getting quite far into the build:

`FAILED: cmake/dependencies/RV_DEPS_GLEW-prefix/src/RV_DEPS_GLEW-stamp/RV_DEPS_GLEW-configure
`

And this as well:

```
rm -f extensions/gl/WGL_*
python bin/parse_xml.py OpenGL-Registry/xml/wgl.xml --api wgl --extensions extensions/gl
bash: line 1: python: command not found
make: *** [Makefile:118: extensions/gl/.dummy] Error 127
ninja: build stopped: subcommand failed.
```

**Reply by geffrak** [reply]
You need a symlink from python3 to python2. `ln -s $(which python3) /usr/local/bin/python` Glew expects `python` to resolve to something. 

Part of the challenge of building OpenRV is that you need to build multiple other projects from source to have it working, so you have to deal with all the _caprices_.

**Comment by rubikstriangle** [comment]
I wrote a script that builds OpenRV on Ubuntu through a Rocky 9 docker. It's an easy single line install.

If you want to give it a try you can find it here:
https://github.com/rubikstriangle/OpenRV-Rocky9-Docker

**Reply by avfxtd** [reply]
Thanks, had a quick look and appears I've missed a crutial step which is installing QT.  Will do this and give your script a try.  Thanks very much!
